lib/charms/mysql/v0/s3_helpers.py

```diff
@@ -93,6 +99,68 @@ def upload_content_to_s3(content: str, content_path: str, s3_parameters: Dict) -
     return True
 
 
+def _read_content_from_s3(content_path: str, s3_parameters: dict) -> str | None:
+    """Reads specified content from the provided S3 bucket.
+
+    Args:
+        content_path: The S3 path from which download the content
+        s3_parameters: A dictionary containing the S3 parameters
+            The following are expected keys in the dictionary: bucket, region,
+            endpoint, access-key and secret-key
+
+    Returns:
+        a string with the content if object is successfully downloaded and None if file is not existing or error
+        occurred during download.
+    """
+    ca_file = None
+    try:
+        logger.info(f"Reading content from bucket={s3_parameters['bucket']}, path={content_path}")
+        session = boto3.session.Session(
+            aws_access_key_id=s3_parameters["access-key"],
+            aws_secret_access_key=s3_parameters["secret-key"],
+            region_name=s3_parameters["region"],
+        )
+        verif = True
+        if ca_chain := s3_parameters.get("tls-ca-chain"):
+            ca_file = tempfile.NamedTemporaryFile()
```

Comment:
suggestion: use a context manager with the NamedTemporaryFile. this will reduce the overload to manually close the file in the finally block